src/components/codeBlock.tsx

```diff
@@ -106,52 +106,154 @@ type KeywordSelectorProps = {
 };
 
 function OrgAuthTokenCreator() {
-  const codeContext = useContext(CodeContext);
+  const [tokenState, setTokenState] = useState<
+    | {status: 'none'}
+    | {status: 'loading'}
+    | {status: 'success'; token: string}
+    | {status: 'error'}
```

Comment:
nit: I would probably define the type `TokenState` outside the component to keep the component cleaner 🤷 
Same goes for the popper options.